src/helper/proxy/index.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @module
+ * Proxy Helper for Hono.
+ */
+
+/*
+ * The following headers will be deleted from the response:
+ * * Content-Encoding
+ * * Content-Length
+ * * Content-Range
+ */
+const forceDeleteResponseHeaderNames = ['Content-Encoding', 'Content-Length', 'Content-Range']
```

Comment:
i'm not sure about `Content-Length` and `Content-Range`, as if the original request is for a byte range, the semantics is not preserved. a probable semantically correct implementation:

1. if the original request contains `Range` header, fast-fail with 400 status code
1. strip `Accept-Range` from response header
2. indicate upstream error if response header contain `Content-Range`

`Content-Length` should be left alone in all case IMO

Reply:
Thank you! This certainly needs fixing.

### Content-Length

As you can see from the results below, in the case of a compressed response, Content-Length is the “size after compression”, so it cannot be returned as is.

```
% curl -I 'https://example.com/'
HTTP/2 200
accept-ranges: bytes
age: 526496
cache-control: max-age=604800
content-type: text/html; charset=UTF-8
date: Wed, 30 Oct 2024 21:13:42 GMT
etag: "3147526947"
expires: Wed, 06 Nov 2024 21:13:42 GMT
last-modified: Thu, 17 Oct 2019 07:18:26 GMT
server: ECAcc (lac/55F5)
x-cache: HIT
content-length: 1256
% curl --compressed --raw -i 'https://example.com/'
HTTP/2 200
content-encoding: gzip
age: 527039
cache-control: max-age=604800
content-type: text/html; charset=UTF-8
date: Wed, 30 Oct 2024 21:07:12 GMT
etag: "3147526947+gzip"
expires: Wed, 06 Nov 2024 21:07:12 GMT
last-modified: Thu, 17 Oct 2019 07:18:26 GMT
server: ECAcc (lac/55AA)
vary: Accept-Encoding
x-cache: HIT
content-length: 648
```

* Delete <- This is the current implementation
* Load the body and reset it <- There is an overhead because you have to clone() and load all the data

Some people may expect the latter, but since hono does not actively add Content-Length to other requests either, I think this is a reasonable response for hono.

However, if there is no `Content-Encoding` in the first place, there is no need to delete the `Content-Length`. I think this should be improved.


### Content-Range

I think the current implementation is incorrect.
It seems that this will not change depending on `Content-Encoding` (although in a real environment, it seems that the result of compression will almost never be returned in a request with Range), so I think it is correct to always return it as is.

Reply:
Fixed in 5998770 and 3a5ef0f.

Reply:
> Content-Length is the “size after compression”, so it cannot be returned as is.

Oh, in that case I am in favor of unconditionally deleting it.

>It seems that this will not change depending on Content-Encoding (although in a real environment, it seems that the result of compression will almost never be returned in a request with Range), so I think it is correct to always return it as is.

I have no reason to believe the new implementation is incorrect and this seems to be a suitable strategy.

Reply:
Thank you!